meteor/sofie-server/model/entities/segment.ts

```diff
@@ -91,11 +90,44 @@ export class Segment {
 		return this.parts
 	}
 
-	public addInfinitePieces(infinitePieces: Piece[]): void {
-		infinitePieces.forEach((piece: Piece) => this.infinitePieces.set(piece.layer, piece))
+	public getFirstSpanningPieceForEachLayerBeforePart(part: Part, layersToIgnore: Set<string>): Piece[] {
+		const indexOfPart: number = this.parts.findIndex((p) => p.id === part.id)
```

Comment:
I know we've talked about having something more descriptive than `p` here, but i can also an argument for the fact that it exists in a pretty well defined context of being a part.

Reply:
Normally I would also call it `part`, but `part` is already taken by the parameter, so I had to call it something else.